quickcheck/src/main/java/fj/test/Arbitrary.java

```diff
@@ -1337,4 +1338,10 @@
     return aa.bind(ab, ac, ad, ae, af, ag, ah,
         a -> b -> c -> d -> e -> f -> g -> h -> p(a, b, c, d, e, f, g, h));
   }
+
+  /**
+   * An arbitrary implementation for naturals.
+   */
+  public static Gen<Natural> arbNatural = arbLong.map(Longs.abs).map(Natural::natural).map(o -> o.some());
```

Comment:
I agree.  In fact, my first implementation of Gen and Shrink for Natural used BigInteger instead of Long.  The problem that I ran into is that the BigInteger Shrink implementation is broken.  There is a call to System.arraycopy in the implementation that throws an exception because the source array is byte[] and the destination array is Byte[].  Once I fixed that, I did some testing of the BigInteger shrink and found that it doesn't appear to shrink correctly.

For example
```
Shrink.shrinkLong.shrink(1000L).take(100).toList();
//Produces List(0,0,500,750,875,938,969,985,993,997,999)
```
versus
```
Shrink.shrinkBigInteger.shrink(BigInteger.valueOf(1000L)).take(100).toList();
//Produces List(0,0,0,512,0,0,0,512,2,2,2,514)
```

I'm not entirely sure how the current implementation is supposed to work so I made an  implementation of Shrink.shrinkBigInteger that uses the exact same logic as Shrink.shrinkLong.  It appears to work correctly although I'm not sure if that is a good approach.  At this point, I think the two options are to go with the Natural Shrink and Gen implementations that uses Long or fix the BigInteger Shrink implementation and then switch the Natural implementations to use BigInteger.